### ISSUE TITLE
Inserting a value with too many decimals throws DbUpdateConcurrencyException

### ISSUE THREAD

**Issue by ixtreon** [dotnet/efcore#17799]
Inserting a `decimal` value with more than 4 decimal places within an SQL Server column of type `money` which is also marked as a concurrency token will throw `DbUpdateConcurrencyException` on follow-up calls to `SaveChanges`. 

The exception message indicates that on consequent calls the context remembers the full-precision value as the "original" one and is thrown when comparing it to the DB value - which is truncated. 

I would expect either a "data would be truncated" exception or silently using the truncated value in the change tracker (similarly to saving into a `datetime` SQL column, I'd imagine?). 

### Steps to reproduce
 - Create a table with a `money` column mapped to an entity and mark it as a concurrency token
 - Save a value with 5 or more decimal places in the `money` column, say "6.66667"
 - Any attempts to further modify, then save the entity throw a `DbUpdateConcurrencyException` indicating the original value (6.66667) is different than the current value in DB (6.6667) 

### Sample Code
``` C#
// entity class
class User
{
    [ConcurrencyCheck] public decimal? Balance { get; set; }
}

// model builder
entity.Property(e => e.Balance).HasColumnType("money");

// sample code
var u = new User { Balance = 1.23456 };
context.GetTable<User>().Add(u);
context.SaveChanges();    // so far so good..

u.Name = "ixtreon";  // omitted from the model for brevity; might not be req'd if we modify "Balance" again (?)
u.SaveChanges();     // DbUpdateConcurrencyException: expected "1.23456" but found "1.2346"
```

### Exception:
```
Unhandled exception during request   [DbUpdateConcurrencyException] : `Database operation expected to affect 1 row(s) but actually affected 0 row(s). Data may have been modified or deleted since entities were loaded. See http://go.microsoft.com/fwlink/?LinkId=527962 for information on understanding and handling optimistic concurrency exceptions.`
     at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ThrowAggregateUpdateConcurrencyException(Int32 commandIndex, Int32 expectedRowsAffected, Int32 rowsAffected)
     at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.ConsumeResultSetWithoutPropagation(Int32 commandIndex, RelationalDataReader reader)
     at Microsoft.EntityFrameworkCore.Update.AffectedCountModificationCommandBatch.Consume(RelationalDataReader reader)
     at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.Execute(IRelationalConnection connection)
     at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(DbContext _, ValueTuple`2 parameters)
     at Microsoft.EntityFrameworkCore.SqlServer.Storage.Internal.SqlServerExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
     at Microsoft.EntityFrameworkCore.Update.Internal.BatchExecutor.Execute(IEnumerable`1 commandBatches, IRelationalConnection connection)
     at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IReadOnlyList`1 entriesToSave)
     at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
     at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
```

### Further technical details

 - If we use `decimal.Round(d, decimals)` beforehand with `decimals < 5` it works fine. 
 - possible duplicate of #16295? At very least, circumstances and stack trace look similar...

EF Core version: 2.2.3
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Framework 4.6.1
Operating system: Win10
IDE: Visual Studio 2019 16.2.5

**Comment by divega** [comment]
Triage: We want to look at how we are deciding the precision of parameters.

**Comment by AndriySvyryd** [comment]
Fixed in 7b489522bb5061018d8e07162ec3acc8c13e3887
We now set precision and scale on parameters
